cpp/src/arrow/pretty_print.h

```diff
@@ -77,6 +77,9 @@ struct PrettyPrintOptions {
 
   /// If true, display schema metadata when pretty-printing a Schema
   bool show_schema_metadata = true;
+
+  /// Delimiter for separating individual elements of an Array (e.g. ",")
+  std::string array_element_delimiter = ",";
```

Comment:
Hmm, good question.

I considered using a shorter name like just `element_delimiter`, but I included `array_` because I thought that if we end up wanting to expose more options like the ability to specify a unique delimiter for `ChunkedArray` elements, that it might become ambiguous. In other words, it might be helpful to have more `array_element_delimiter` and `chunked_array_element_delimiter`.

Do you think we should still remove `array_` given this consideration?

Reply:
For `ChunkedArray`, I think that `array_delimiter` is better.
Because `ChunkedArray`'s element is an `Array`. `array_` is more concrete than `element_` for `ChunkedArray`.

Reply:
Interesting.

Then how about the following? I want to try simplifying your idea.

```cpp
struct PrettyPrintDelimiters {
  std::string open = "[";
  std::string close = "]";
  std::string element = ",";
};

struct PrettyPrintOptions {
  // ...
  PrettyPrintDelimiters array_delimiters = PrettyPrintDelimiters::Defaults();
  PrettyPrintDelimiters chunked_array_delimiters = PrettyPrintDelimiters::Defaults();
  // ...
};
```

Reply:
Ok, I've implemented the `PrettyPrintDelimiters` approach as suggested.

I intentionally chose not to add `open` or `close` yet because my plan is to follow up with a separate PR to address this since that functionality is also captured by the separate issue https://github.com/apache/arrow/issues/37979. As soon as this is merged in, I will work on adding `open` and `close`.

Reply:
Marking this as resolved.